### ISSUE TITLE
CodeLens support broken on electron

### ISSUE THREAD

**Issue by martin-fleck-at** [eclipse-theia/theia#10879]
### Bug Description:
It seems to me that code lens support is broken somehow. I tried to add a codelens provider using `monaco.languages.registerCodeLensProvider` but the respective `provideCodeLenses` is never called. I am using electron and I believe that already was working some time ago. 

### Steps to Reproduce:
1. I created a branch on top of the current master: https://github.com/martin-fleck-at/theia/tree/codelens-test with the following commit https://github.com/martin-fleck-at/theia/commit/78c2c18ae4c0731c282f7293e4691947c8e95410. This adds a hover provider for the first line of any (monaco-managed) file and a code lens provider that should also decorate the first line.
2. Open any file in any workspace that has more than one line.
3. You should be able to see the code lens. However, the method is never triggered. The hover provider works as expected.

### Additional Information
- Operating System: Ubuntu
- Theia Version: current master (using electron)

**Comment by martin-fleck-at** [comment]
@msujew What is the extension you are using to show the references or should that already be present by default?

**Comment by msujew** [comment]
@martin-fleck-at It's the default JS/TS built-in extension. You can turn on code lenses for references in the settings of that extension.

**Comment by martin-fleck-at** [comment]
@msujew Thank you very much for your help! It seems they are not working for me. So it might be that a problem on my end. Now I also see a lot of `"Invalid cast. The supplied value [object Object] did not pass the test 'isArrayLiteralExpression'."` errors in typescript.js so that is definitely not good ;-) I just cleaned my repository and all settings and I still see the same errors, maybe it is related to that. Have you ever seen that by any chance? And which node version are you using? Unfortunately, I need to leave now but I'll see if I can find out more tomorrow. If it is all on my end, I'll close this bug if that is alright.

**Comment by msujew** [comment]
@martin-fleck-at Sure, if you are somehow able to fix the issue on your side, feel free to close it. 

I'm running Node 14.18.1, but I've also tested it with 12.20.1. Interesting, I haven't seen those errors yet. Does the missing code lens issue also exist when you build the browser version?

**Comment by martin-fleck-at** [comment]
@msujew Thank you! I tried with several node versions and both the browser and the electron backend and I always see the same errors for typescript. They happen when the JS/TS language features are being initialized. So while this is definitely an issue, I am not sure it is the actual cause of my lenses not showing up since my example should show the lenses on any file. When opening a non-typescript file I do not see the same errors but my lenses still do not show up.

Just for an additional test, I also installed the Java extension as it also has some lens support and that one also does not show up for me. I am really not sure what else to try. If it is not too much to ask, could you please test my branch to see if it works for you and the `provideCodeLenses` method is called?

**Comment by msujew** [comment]
@martin-fleck-at Sure, I'll look into it and get back to you 👍

**Comment by msujew** [comment]
@martin-fleck-at I can confirm that it does not work with your changes applied, I get this error in the console when opening a file:

![grafik](https://user-images.githubusercontent.com/4377073/158386147-1a4955c7-9340-44cf-82d4-689cf6a558e1.png)

Seems like something inside of Monaco isn't happy with your changes.

**Comment by martin-fleck-at** [comment]
@msujew Thank you very much for your help! I used Gitpod to avoid any potential workspace problems and I saw the same problem you did and managed to solve the issue you mentioned - my typescript issues still persist but are unrelated. Unexpectedly for me but interestingly, it seems that monaco creates a hash over all code lens providers together with the model id so that requests can be debounced according to the provider performance:
```
codelensController.ts > CodeLensContribution._onModelChange > 195: this._getCodeLensModelDelays.get(model) >
languageFeatureRegistry.ts > LanguageFeatureRequestDelays.get(ITextModel) > _key(ITextModel) > 197: has(this.registry.all(model)) >
hash.ts > doHas(obj) > tries to has all properties including EditorManager
```

In my implementation of the code lens provider I injected the EditorManager as a property and creating a hash over that lead to the `too much recursion` error we saw. Simply pulling out the codelens provider into a separate class solved the issue and everything is working as expected:

![image](https://user-images.githubusercontent.com/19170971/158576566-c98413dc-08ea-4fbc-9a3b-6ff016f35169.png)

Thank you again for your help and I now happily close this bug ;-)
